### PR TITLE
[Form][FrameworkBundle] Move FormPass to the Form component

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

So that anyone using only Form and DI can use it for registering form types/type guessers.
Follows #19443, related to #21284

### CODE REVIEW

**Comment by dunglas** [comment]
👍

File: src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php [hunk 1]
```diff
@@ -83,7 +83,9 @@ public function build(ContainerBuilder $container)
         if (class_exists(AddConsoleCommandPass::class)) {
             $container->addCompilerPass(new AddConsoleCommandPass());
         }
-        $container->addCompilerPass(new FormPass());
+        if (class_exists(FormPass::class)) {
```
Comment (sstok) [inline src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php]: As this going to be very big would it make sense to use a private helper method?
`$this->addExistingCompilerPass($className, $arguments = array())`.

So you don't end-up with numerous if-statements 👍

Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php]: I tend to agree 👍  let's see if deciders think it's worth it

Comment (dunglas) [inline src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php]: A simple private method in this extension can do the trick isn't it?

Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php]: Sure, method added

**Comment by chalasr** [comment]
Updated this to make the form extension service id and form type/type_extension/type_guesser tags configurable through the constructor, defaulting to the ones used in the framework.

**Comment by fabpot** [comment]
Thank you @chalasr.

**Comment by ro0NL** [comment]
> This needs a second review :)

Well.. the approach now differs with #20250 (see discussion https://github.com/symfony/symfony/pull/20250#discussion_r84143705, ie. default args vs. required args)

Yet im rethinking it.. maybe having defaults in library code is good for convenience any way, and #20250 needs an update.

**Pull request merged**